### PR TITLE
updates for Sphinx 1.5 and 1.6

### PR DESCRIPTION
**Pull request by newville**
This tries to address some problems using Sphinx 1.5 and 1.6.3

I still cannot build latexpdf yet, and there are some problems with sub-figure numbering/lettering, but this at least builds html.

### CODE REVIEW

**Comment by newville** [comment]
this now gives properly numbered major.minor numbers for figures, but does not correctly include the letter in the captions for sub-figures.  Also, latex is produced but is incomplete (conversion of *.RST to *.in.tex file seems to be missing).   

Any chance of help on fixing these items?

**Comment by newville** [comment]
OK, this now fixes (sort of kludgy...) figure and sub-figure labeling for HTML. 

I have not looked into the latex generation problems in detail.

**Comment by jterrace** [comment]
Looks like make html is failing on Travis CI? ImportError: cannot import name educate_quotes_latex

File: extensions/numfig.py [hunk 1]
```diff
@@ -78,6 +78,17 @@ def doctree_read(app, doctree):
     env.docnames_by_figname = docnames_by_figname
     env.docname_figs = docname_figs
 
+def figname_is_id(s):
+    "determine if string is 'idX' where X is an integer"
```
Comment (jterrace) [inline extensions/numfig.py]: was this supposed to be triple quotes?

**Comment by newville** [comment]
@jterrace Hi, and sorry but this PR and the problem at hand are both a bit messy:

First, yes, `educate_quotes_latex()` from smartypants  has been removed from sphinx.  I added it back into `latex_mods.py` for now.  It now passes on Python 2.7, but not 2.6.  Sphinx no longer supports Python 2.6,  so I removed travis testing of 2.6.

But also, the numbering / lettering for sub-figures is pretty broken with sphinxtr and sphinx v 1.5 and 1.6.  The kludge here is not robust.

It seems that many aspects of sphinxtr have been incorporated into recent versions of sphinx, and what remains is incompatible.   For subfig and numfig, I believe the best way forward would be to use sphinx's own numfig option, modifying that to support sub-figures.

I'm afraid that I know very little of the internals of docutils or sphinx.   But, I really really want good citations as with natbib, numbered sections, numbered figures, and sub-figures.  

FWIW, I'm not writing a Thesis, but documentation for scientific software (see http://xraypy.github.io/xraylarch/). Being a Latex fan from longer ago than I care to admit, the  HTML result is more important for this need than the latex backend.

**Comment by jterrace** [comment]
Yeah, I mean things are already broken, so this is only improving things :)

**Pull request merged**